### PR TITLE
Upgrade: Add `--shadow-inner` theme variable when `shadow-inner` class is used

### PR DESCRIPTION
**Pull request by philipp-spiess**
We are dropping the default `--shadow-inner` theme variable from Tailwind CSS v4 in favor of the `inset-shadow` utilities. We can't however, easily codemod usages from `shadow-inner` to `insert-shadow-sm` because of how this utility interacts with shadow colors. E.g.:

```diff
- <div class="shadow-inner shadow-red-500">
+ <div class="inset-shadow-sm inset-shadow-red-500">
```

Since we can't do codemods on connected candidates right now, we instead backfill support for `shadow-inner` if it is used in one of your templates files. This is as simple as adding the following CSS theme variable:

```css
@theme {
  --shadow-inner: inset 0 2px 4px 0 rgb(0 0 0 / 0.05);
}
```

## Test plan

Added an integration test for a setup using `shadow-inner` in its templates.

### CODE REVIEW

File: packages/@tailwindcss-upgrade/src/template/migrate.ts [hunk 1]
```diff
@@ -85,12 +80,14 @@ export default async function migrateContents(
...
-    await migrateContents(designSystem, userConfig, contents, extname(file)),
-  )
-}
+//   await fs.writeFile(
+//     fullPath,
+//     await migrateContents(designSystem, userConfig, contents, extname(file)),
+//   )
+
+// let candidates = await extractRawCandidates(contents, extension)
+// }
```
Comment (thecrypticace) [inline packages/@tailwindcss-upgrade/src/template/migrate.ts:93]: You can delete this

File: packages/@tailwindcss-upgrade/src/index.ts [hunk 2]
```diff
@@ -142,27 +189,20 @@ async function run() {
...
+        for (let [file, { candidates, contents }] of candidatesByFile.entries()) {
+          let newContents = await migrateTemplateContents(
+            candidates,
+            config.designSystem,
+            config.userConfig,
+            contents,
+          )
+
+          if (newContents !== contents) {
+            await fs.writeFile(file, newContents)
```
Comment (thecrypticace) [inline packages/@tailwindcss-upgrade/src/index.ts:203]: Only thing to note here is this is no longer happening in parallel. Probably not a big deal but at least wanted to point it out.

**Comment by adamwathan** [comment]
Can delete `--shadow-inner` from the default theme as part of this PR too 👍🏻

File: packages/@tailwindcss-upgrade/src/index.ts (hunk 2 above)
Comment (philipp-spiess) [reply packages/@tailwindcss-upgrade/src/index.ts:203]: Oh you're right, I only made the file reads parallel. Should do it here too 👍

**Comment by philipp-spiess** [comment]
@tailwindlabs/engineering With `--shadow-inner` moving into the _deprecated_ section of the `@theme`, I guess this PR is no longer needed. Does anyone have any objections?

**Pull request closed without merging**